FaunaDB.Client.Test/TimeoutClientTest.cs

```diff
@@ -0,0 +1,116 @@
+﻿using FaunaDB.Client;
+using System.Diagnostics;
+using FaunaDB.Types;
+using System;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using System.Net;
+using System.Net.Sockets;
+
+using static FaunaDB.Query.Language;
+
+namespace Test
+{
+#if NETFRAMEWORK
+
+    [TestFixture]
+    public class TimeoutClientTest
+    {
+        private DummyServer dummy;
+        private FaunaClient client;
+
+        [SetUp]
+        public void SetUp()
+        {
+            dummy = new DummyServer();
+            dummy.Start();
+
+            client = new FaunaClient("secret", "http://127.0.0.1:9999", timeout: TimeSpan.FromSeconds(10));
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            dummy.Stop();
+        }
+
+        [Test]
+        public void TestServerTimingOut()
+        {
+            Stopwatch watch = new Stopwatch();
+            watch.Start();
+
+            Task<Value> task0 = client.Query(Now());
+
+            Assert.ThrowsAsync<TimeoutException>(async () => await task0);
+
+            watch.Stop();
+
+            Assert.IsTrue(task0.IsCompleted, "the task0 is completed");
+            Assert.IsTrue(task0.IsFaulted, "the task0 failed");
+
+            StringAssert.Contains("The operation has timed out.", task0.Exception.Message);
+
+            Assert.IsTrue(watch.Elapsed.TotalSeconds >= 10 && watch.Elapsed.TotalSeconds <= 12);
```

Comment:
I'm not sure about that assertion. Maybe its overkill at this point, but the older impl. will take ~60s to cancel the task, throwing a `TaskCanceledException` instead of `TimeoutException`